### DISCUSSION TITLE
cpg0014-jump-adipocyte

### DISCUSSION THREAD

**Discussion by ErinWeisbart** [broadinstitute/cellpainting-gallery#58] [Dataset descriptions]
Segmentation/ Feature extraction is being performed by Erin in Cimini lab
Profile creation is being performed by Erin in Cimini lab
Data can be public in RODA Immediately

Update as generated:  
profiler repo = https://github.com/jump-cellpainting/cpg0014-JUMP-adipocyte
[Link to publication repo]  
cellpainting-gallery identifier = cpg0014-jump-adipocyte

- [ ] Metadata collection form filled out by collaborator or Imaging Platform member (https://airtable.com/shrVxz9DcoMlDoCBI)
- [ ] Metadata completely filled out in Project Profiler Database (Imaging Platform internal use only)
- [x] Segmentation/Feature extraction complete
- [ ] Profiling complete

Transfer to CellPainting Gallery:
- [x] Upload data to RODA (is private by default)
- [ ] Run validation script to ensure completion (https://github.com/jump-cellpainting/data-validation)
- [ ] Update cellpainting-gallery/README.md
- [ ] Make RODA entry public

If data is being published, prepare for publication:
- [ ] Run Distributed-BioFormats2Raw to create .ome.zarr files
- [ ] Upload (meta)data to IDR (images remain hosted in cellpainting-gallery).

Once published:
- [ ] Make IDR entry public
- [ ] Update cellpainting-gallery/README.md and open-data-registry/cellpainting-gallery.yml to reflect publication

**Comment by shntnu** [comment]
@ErinWeisbart 

(1) 
This dataset will have at least two different sets of CellProfiler-derived profiles (the second one processed using the JUMP recipe)
Context: https://github.com/jump-cellpainting/cpg0014-jump-adipocyte-data/issues/2#issuecomment-2638117918

Any thoughts on how we should handle this, nomenclature-wise?

(2)

Do you know why there are DVC markers here? Presumably because it was all synced up from the GitHub data repo?

<img width="713" alt="image" src="https://github.com/user-attachments/assets/b2992a46-2cb1-43d9-be79-66f05dca6247" />

Note that the DVC backend is in a separate folder

<img width="552" alt="image" src="https://github.com/user-attachments/assets/6a5a0f8a-1bb6-400d-a223-8f8e90f0676d" />

**Reply by ErinWeisbart** [reply]
1) Hmmm.. We have a precedent of reprocessed datasets having their original profiles being named `profiles_orig` and the new profiles going to `profiles`, but those datasets were historical and the new `profiles` matched new `analysis`. Am I correct in understanding that this is using the same `analysis` but creating profiles differently so that it matches JUMP? And these new profiles come off of the same PLATE_augmented.csv? If that's the case, my intuition would say they would go in the same `profiles` folder but have different suffixes on the profile filenames.

2) Yeah, I would guess it was a sync from a repo. We should delete .dvc files in the `profiles` folder

**Reply by shntnu** [reply]
> We have a precedent of reprocessed datasets having their original profiles being named profiles_orig and the new profiles going to profiles, but those datasets were historical and the new profiles matched new analysis. 

Thanks for the recap

> Am I correct in understanding that this is using the same analysis but creating profiles differently so that it matches JUMP? 

Yes

> And these new profiles come off of the same PLATE_augmented.csv? 

Yes

> If that's the case, my intuition would say they would go in the same profiles folder but have different suffixes on the profile filenames.

Ignore the details below and skip over to the next comment

<details>

Instead, maybe we could borrow from the way we have structured [`cpg0016-jump-assembled/source_all/workspace/profiles/`](https://cellpainting-gallery.s3.amazonaws.com/index.html#cpg0016-jump-assembled/source_all/workspace/profiles/)

Here's the structure of that folder

```
jump-profiling-recipe_2024_a917fa7
└── CRISPR
         └── profiles_wellpos_cc_var_mad_outlier_featselect_sphering_harmony_PCA_corrected
                   └── profiles_wellpos_cc_var_mad_outlier_featselect_sphering_harmony_PCA_corrected.parquet <- single parquet for whole CRISPR dataset
```

Proposed structure under `cpg0016-jump/source_4/workspace/profiles_alternative/` (taken as an example):

```
jump-profiling-recipe_2024_a917fa7
         └── profiles_wellpos_cc_var_mad_outlier_featselect_sphering_harmony_PCA_corrected
                    └── 2021_04_26_Batch1
                             └── BR00117035
                                       └── (parquet files)
```

For comparison, if we had used this structure in `profiles/`:

```
profiling-recipe_<hash> <- identifier of the recipe
         └── <configuration_id> <- identifier of the config file
                    └── 2021_04_26_Batch1
                             └── BR00117035
                                       └── (parquet files)
```

Does that makes sense?

For those who might read this without more context: we are trying to stick with a standardized folder structure for usability, but it has limitations because capturing provenance is very hard done this way.

</details>

**Reply by shntnu** [reply]
Oh wait, it turns out that the output of the jump-profiling-recipe for `cpg0014-jump-adipocyte` is structured exactly the same way as it is for `cpg0016-jump-assembled` (which makes sense):

![image](https://github.com/user-attachments/assets/fd3ac2ff-4b8f-4ad6-ac60-1d3b491b100f)

See https://github.com/jump-cellpainting/cpg0014-jump-adipocyte-data/issues/2#issuecomment-2641833280

Ok so in that case we should do things nearly identical to `cpg0016-jump-assembled`, except that we should also upload the other artifacts (`metrics`, `stats`) for completeness.

```
cpg0014-jump-adipocyte/broad/workspace/profiles_assembled/jump-profiling-recipe_2025_3aa6b25
└── ALL
    └── metrics <- phenotypic activity metrics
    ├── stats
    │   ├── norm <- files for normalizing the data
    │   └── sphering <- intermediate files related to sphering; we should drive this down
    └── profiles
      └── profiles_wellpos_cc_var_mad_outlier_featselect_sphering_harmony
          ├── profiles_wellpos_cc_var_mad_outlier_featselect_sphering_harmony.parquet
          ├── profiles_wellpos_cc_var_mad_outlier_featselect_sphering.parquet
          ├── profiles_wellpos_cc_var_mad_outlier_featselect.parquet
          ├── profiles_wellpos_cc_var_mad_outlier.parquet
          ├── profiles_wellpos_cc_var_mad.parquet
          ├── profiles_wellpos_cc_var.parquet
          ├── profiles_wellpos_cc.parquet
          ├── profiles_wellpos.parquet
          └── profiles.parquet
```
      
 What do you think @ErinWeisbart? I am wary of creating an entirely new structure :( But `jump-profiling-recipe` is structured differently so -- until we refactor that -- we'd need to go with a solution like this. We will mark this structure as experimental and then revise it when we've settled on the output structure of `jump-profiling-recipe`.

**Reply by shntnu** [reply]
For now, I will sync just the profiles and metrics so that we can keep moving in that project

```
s5cmd sync "/dgx1nas1/storage/data/shared/cpg00014_adipocytes/jump-profiling-recipe/orf_adipocytes/metrics/*.parquet" s3://cellpainting-gallery/cpg0014-jump-adipocyte/broad/workspace/profiles_assembled/jump-profiling-recipe_2025_3aa6b25/ALL/profiles_wellpos_cc_var_mad_outlier_featselect_sphering_harmony/metrics/

s5cmd sync "/dgx1nas1/storage/data/shared/cpg00014_adipocytes/jump-profiling-recipe/orf_adipocytes/*.parquet" s3://cellpainting-gallery/cpg0014-jump-adipocyte/broad/workspace/profiles_assembled/jump-profiling-recipe_2025_3aa6b25/ALL/profiles_wellpos_cc_var_mad_outlier_featselect_sphering_harmony/profiles/
```
